code/modules/xenoarchaeology/tools/xenoarchaeology_machines.dm

```diff
@@ -0,0 +1,323 @@
+	ADD_TRAIT(src, TRAIT_ARTIFACT_IGNORE, GENERIC_ITEM_TRAIT)
+
+/obj/machinery/xenoarchaeology_machine/attackby(obj/item/I, mob/living/user, params)
+	var/list/modifiers = params2list(params)
+	var/atom/target = get_target()
+	//Prechecks
+	if(move_inside && length(held_contents) >= max_contents)
+		return
+	///Move the item to our target, so we can work with it, like we're a table
+	if(user.a_intent != INTENT_HARM && !(I.item_flags & ABSTRACT))
+		if(user.transferItemToLoc(I, target, silent = FALSE))
+			//Center the icon where the user clicked.
+			if(!LAZYACCESS(modifiers, ICON_X) || !LAZYACCESS(modifiers, ICON_Y))
+				return
+			//Clamp it so that the icon never moves more than 16 pixels in either direction (thus leaving the table turf)
+			I.pixel_x = clamp(text2num(LAZYACCESS(modifiers, ICON_X)) - 16, -(world.icon_size/2), world.icon_size/2)
+			I.pixel_y = clamp(text2num(LAZYACCESS(modifiers, ICON_Y)) - 16, -(world.icon_size/2), world.icon_size/2)
+			//Handle contents
+			if(move_inside)
+				register_contents(I)
+	else
+		return ..()
+
+/obj/machinery/xenoarchaeology_machine/attack_hand(mob/living/user)
+	. = ..()
+	empty_contents()
+
+/obj/machinery/xenoarchaeology_machine/proc/register_contents(atom/atom_target)
+	RegisterSignal(atom_target, COMSIG_PARENT_QDELETING, PROC_REF(unregister_contents))
+	RegisterSignal(atom_target, COMSIG_MOVABLE_MOVED, PROC_REF(unregister_contents))
+	held_contents += atom_target
+
+/obj/machinery/xenoarchaeology_machine/proc/unregister_contents(datum/source)
+	SIGNAL_HANDLER
+
+	held_contents -= source
+	UnregisterSignal(source, COMSIG_PARENT_QDELETING)
+	UnregisterSignal(source, COMSIG_MOVABLE_MOVED)
+
+/obj/machinery/xenoarchaeology_machine/proc/get_target()
+	return move_inside ? src : drop_location()
+
+/obj/machinery/xenoarchaeology_machine/proc/empty_contents(atom/movable/target, force)
+	if(target && (list(target) & held_contents || force))
+		target.forceMove(get_turf(src))
+		unregister_contents(target)
+		return
+	for(var/atom/movable/AM in held_contents)
+		AM.forceMove(get_turf(src))
+		unregister_contents(AM)
+
+//Circuitboard
+/obj/item/circuitboard/machine/xenoarchaeology_machine
+	name = "place holder (Machine Board)"
+	icon_state = "science"
+	build_path = /obj/machinery/xenoarchaeology_machine
+	req_components = list(
+		/obj/item/stack/ore/bluespace_crystal = 5,
+		/obj/item/stock_parts/matter_bin = 1)
+
+/*
+	Scale, measures artifact weight
+*/
+/obj/machinery/xenoarchaeology_machine/scale
+	name = "industrial scale"
+	desc = "A piece of industrial equipment, designed to weigh thousands of kilograms."
+	icon_state = "scale"
+	circuit = /obj/item/circuitboard/machine/xenoarchaeology_machine/scale
+
+/obj/machinery/xenoarchaeology_machine/scale/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>Interact to measure artifact weight.\nLabeled artifacts will also show label weights, against the total.</span>"
+
+/obj/machinery/xenoarchaeology_machine/scale/attack_hand(mob/living/user)
```

Comment:
instead of defining `attack_hand`, it should have a proc `/obj/machinery/xenoarchaeology_machine/proc/handle_machine_feature(mob/living/user)` and each subtype of the xeno research machine parent should have this.

Reply:
naur :(

Reply:
you should

Reply:
I dont really see the point of a snowflake proc ngl